### PR TITLE
Change migration format

### PR DESCRIPTION
**Pull request by pelmered**
I've been having issues with this package when I use autoloading of migrations from domain paths in my project that uses DDD, in specific this package: [lunarstorm/laravel-ddd](https://github.com/lunarstorm/laravel-ddd). 
I get error like `Cannot declare class CreateUsersIndex, because the name is already in use` because the file is loaded multiple times which makes the class names clash. That DDD package autoloads several types of files which scans the Domain directory and creates this problem. 

While this could be fixed in that package as well, but I think it would be better if this migration format is updated to be the same as the Laravel format with anonymous classes. 
I.e: 
`return new class implements MigrationInterface`
Instead of:
`final class CreateVehiclesIndex implements MigrationInterface`
The file names are not changed. 

This change is made to be backwards compatible with the current format. 

I have added some tests, but some more tests could probably be added to to test this format further.

### CODE REVIEW

File: tests/migrations/2024_04_08_113433_test_new_format.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+
+use Elastic\Adapter\Indices\Mapping;
+use Elastic\Adapter\Indices\Settings;
+use Elastic\Migrations\Facades\Index;
+use Elastic\Migrations\MigrationInterface;
+
+final class TestNewFormat implements MigrationInterface
```
Comment (babenkoivan) [inline tests/migrations/2024_04_08_113433_test_new_format.php:9]: Shouldn't it be `return new class implements MigrationInterface`?

**Comment by babenkoivan** [comment]
Hey @pelmered, thank you for the PR! Could you please resolve the conflicts and look at the comment I left?

File: tests/migrations/2024_04_08_113433_test_new_format.php (hunk 1 above)
Comment (pelmered) [reply tests/migrations/2024_04_08_113433_test_new_format.php:9]: Yes, you are right! I'll fix that.

File: src/Factories/MigrationFactory.php [hunk 2]
```diff
@@ -10,11 +10,23 @@ class MigrationFactory
...
+        if (is_object($class)) {
+            // This is an annonymous class so return it directly
+            return $class;
+        }
+
+        try {
+            $className = Str::studly(implode('_', array_slice(explode('_', $file->name()), 4)));
+            /** @var MigrationInterface $migration */
+            $migration = resolve($className);
+        } catch (\Illuminate\Contracts\Container\BindingResolutionException) {
```
Comment (babenkoivan) [inline src/Factories/MigrationFactory.php:24]: Do you think this is needed? File names should be unique, how can we get in a situation when `require_once` is called twice?

**Comment by babenkoivan** [comment]
Could you please also adjust the code so that all checks would be green?

**Pull request closed without merging**